ledger/ledgercore/statedelta.go

```diff
@@ -113,32 +113,44 @@ type NewBalanceRecord struct {
 	AccountData
 }
 
-// AppParamsRecord represents app params in deltas
-type AppParamsRecord struct {
-	Aidx   basics.AppIndex
-	Addr   basics.Address
-	Params *basics.AppParams
+// AssetHoldingDelta records a changed AssetHolding, and whether it was deleted
+type AssetHoldingDelta struct {
+	Holding *basics.AssetHolding
+	Deleted bool
 }
 
-// AssetParamsRecord represents asset params in deltas
-type AssetParamsRecord struct {
-	Aidx   basics.AssetIndex
-	Addr   basics.Address
-	Params *basics.AssetParams
+// AssetParamsDelta tracks a changed AssetParams, and whether it was deleted
+type AssetParamsDelta struct {
+	Params  *basics.AssetParams
+	Deleted bool
+}
+
+// AppLocalStateDelta tracks a changed AppLocalState, and whether it was deleted
+type AppLocalStateDelta struct {
+	State   *basics.AppLocalState
+	Deleted bool
 }
 
-// AppLocalStateRecord represents app local state in deltas
-type AppLocalStateRecord struct {
-	Aidx  basics.AppIndex
-	Addr  basics.Address
-	State *basics.AppLocalState
+// AppParamsDelta tracks a changed AppParams, and whether it was deleted
+type AppParamsDelta struct {
+	Params  *basics.AppParams
+	Deleted bool
 }
 
-// AssetHoldingRecord represents asset holding in deltas
-type AssetHoldingRecord struct {
+// AppResourceRecord represents AppParams and AppLocalState in deltas
+type AppResourceRecord struct {
+	Aidx   basics.AppIndex
+	Addr   basics.Address
+	Params AppParamsDelta
```

Comment:
maybe `Params` -> `GlobalParams`, and `State` -> `LocalState` ?

Reply:
I can change State to LocalState but I was aiming for using only a "words after App/Asset" naming convention like Holding, Params, etc.